app/service_invite/rest.py

```diff
@@ -121,3 +124,95 @@ def validate_service_invitation_token(token):
 
     invited_user = get_invited_user_by_id(invited_user_id)
     return jsonify(data=invited_user_schema.dump(invited_user)), 200
+
+
+@service_invite.route("/service/<service_id>/invite/request-for/<user_to_invite_id>", methods=["POST"])
+def request_user_invite(service_id, user_to_invite_id):
+    request_json = request.get_json()
+    user_requesting_invite = get_user_by_id(user_to_invite_id)
+    recipients_of_invite_request_ids = request_json["service_managers_ids"]
+    recipients_of_invite_request = [get_user_by_id(recipient_id) for recipient_id in recipients_of_invite_request_ids]
+    service = dao_fetch_service_by_id(service_id)
+    reason_for_request = request_json["reason"]
+    invite_link_host = request_json["invite_link_host"]
+    accept_invite_request_url = f"{invite_link_host}/services/{service.id}/users/invite/{user_requesting_invite.id}"
+
+    # # Ensure that the user making the request is already not part of the service
+    if user_requesting_invite.services and service in user_requesting_invite.services:
+        message = f"You are already a member of {service.name}"
+        raise BadRequestError(message=message)
+
+    # Send the user's service invite request to the service managers listed
+    send_service_invite_request(
+        user_requesting_invite, recipients_of_invite_request, service, reason_for_request, accept_invite_request_url
+    )
+
+    # Send a receipt email to the user that requested the invite
```

Comment:
[nit]: I don't think we need these comments because you've given the functions nice sensible names :)

Reply:
I erred on the side of leaving in the comments for the review, I will take them out.